### DISCUSSION TITLE
Consider heat losses in pipe component

### DISCUSSION THREAD

**Discussion by HaSchneider** [oemof/tespy#547] [Ideas]
## Background 
For models considering large pipe networks with relatively small diameters like district heating or steam networks the heat losses of the pipes could be crucial. The current pipe components cant handle that sufficient right now, as it is basicaly the heat exchanger class, where only an area independent heat transfer coefficient can be used. It is thereby not possible to perform senstivity studies on network topologies and calculation of the maximum realisable pipe lengths regarding the temperature losses. 

## Idea
Approve the pipe component by adding the opportunity to calculate the heat losses over the pipe by calculating an area dependent heat transfer coefficient based on definition of pipe length, insulation, pipe diameter, ...

## Code basis
The code of [DWSIM ](https://github.com/DanWBR/dwsim/blob/windows/DWSIM.UnitOperations/UnitOperations/Pipe.vb#L1380-L1695) for calculating the overall heat transfer coefficient of a pipe could be used as basis: `CalcOverallHeatTransferCoefficient`

**Comment by fwitte** [comment]
Hi @HaSchneider,

thanks for reaching out on this. would you be interested to implement this. I do not think that it would be too difficult, I can help you to get started with it :). If you are interested we can set up a meeting!

Best

Francesco

**Reply by HaSchneider** [reply]
Yes, I can work on that! I am just not that familiar with the tespy structure and basic setup of components. But I would appreciate if we could meet for that!

**Reply by fwitte** [reply]
I am out of office, right now, but you can already send me an email: francesco.witte (at) dlr.de. will come back to you next week
